packages/cli/src/cmds/preview/preview.ts

```diff
@@ -63,27 +70,44 @@ export default class Preview extends YargsCommand {
     [argName: string]: boolean | string | string[] | undefined;
   }): Promise<Result<null, FxError>> {
     if (args.local && args.remote) {
-      return err(errors.ExclusiveLocalRemoteOptions());
+      cliTelemetry.sendTelemetryEvent(TelemetryEvent.PreviewStart);
+      const error = errors.ExclusiveLocalRemoteOptions();
+      cliTelemetry.sendTelemetryErrorEvent(TelemetryEvent.Preview, error);
+      return err(error);
     }
 
+    const previewType = args.local || (!args.local && !args.remote) ? "local" : "remote";
+    this.telemetryProperties[TelemetryProperty.PreviewType] = previewType;
+
     const workspaceFolder = path.resolve(args.folder as string);
```

Comment:
You need to enable RootFolder for preview command telemetries, I think. This is to make sure we can get some config context in telemetry like "appid". Do something like this: CliTelemetry.withRootFolder(rootFolder).sendTelemetryEvent(TelemetryEvent.ProvisionStart);

Reply:
Sure. Updated to use `cliTelemetry.withRootFolder`.